### ISSUE TITLE
`eth.*` functions should accept an ID of the chain

### ISSUE THREAD

**Issue by jbargu** [paralink-network/paralink-node#19]
With the new addition of multiple EVM chains we should update the `eth.balance`, `eth.function` calls in PQL to specify the chain name/ID. We should figure out how to uniquely define a chain ID in the PQL to be consistent across the PQLs from different nodes.

**Comment by frisitano** [comment]
I believe as of [EIP-155](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-155.md) EVM based chains have[ _official_ ID's](https://chainid.network/).  Do you think this may be a suitable standard to adhere to?

**Comment by frisitano** [comment]
I've done a POC using the chain reference data from above, you can see the compare [here](https://github.com/paralink-network/paralink-node/compare/master...frisitano:chain_id).

**Comment by jbargu** [comment]
Hmm, I see some of Polkadot projects but not all (e.g. Plasm and Moonbeam are not present). I would maybe rather use the explicit string ID without using magic numbers. I believe concatenation of `shortName` + `network` is unique, therefore if the user wants to query mainnet, he writes `eth.mainnet`, if he wants Kovan testnet, he writes `kov.kovan` if he wants Acala: `aca.mainnet`. I would then use the data in `chains.json` to check whether the EVM network that was initialized in the node complies with it. So whenever the node user creates an `evm = EvmChain(name='kov.kovan')` through the `chain_config.json` a call to `evm.get_connection()` would check if the chain ID/network ID corresponds to the one found in the `chains.json` (chainId: 42, networkId: 42) and raise an exception if it does not.

For the networks that are not listed in the `chains.json` and are of unknown source we omit the check. For example if the user wants to run a local dev node, he can create his own `foo.testnet` and query it through his node. If we opted for the magic numbers, the user would then have to take care of having unique network ID, which is then additional layer of configuration.

For the networks that are not listed but are supported from our side (e.g. Plasm, Moonbeam), we add the entry to the `chain.json` ourselves.

I am not sure if it is possible for `shortName` + `network` to be duplicated but in that case another chainId identifier could be added such as `eth.mainnet.1`, `eth.mainnet.2`.

Thank you for your research and your work, feel free to poke a hole into the above!

**Comment by frisitano** [comment]
I agree with the points you've made.  I have created PR #26 with a draft implementation, it is ready for a review at your convenience.

I have included chain reference data for Moonbase Alpha (Moonbeam TestNet) in `chains.json` but was unable to find suitable data for Plasm.

As for your point regarding the potential duplication of `shortName` + `network` my search was not able to find any examples of this.  It may be prudent to include the chainId but I will take your guidance on this.
